tiledb/sm/query/query.cc

```diff
@@ -2013,28 +2023,44 @@ Status Query::set_est_result_size(
 }
 
 Status Query::set_layout(Layout layout) {
-  if (type_ != QueryType::READ && type_ != QueryType::WRITE &&
-      type_ != QueryType::MODIFY_EXCLUSIVE) {
-    return LOG_STATUS(Status_SerializationError(
-        "Cannot set layout; Unsupported query type."));
-  }
+  switch (type_) {
+    case (QueryType::READ):
+      if (status_ != QueryStatus::UNINITIALIZED) {
+        return logger_->status(
+            Status_QueryError("Cannot set layout after initialization"));
+      }
```

Comment:
@KiterLuc Should this status error also be returned for the other query types when the layout is not global order?